src/components/ImportRecordsFlow/stepThree/__test__/stepThree.spec.js

```diff
@@ -25,22 +31,60 @@ describe('<StepThree />', () => {
     });
     it('should pass params to modal', () => {
         const component = mount(<StepThree {...baseParams} />);
-        const modal = component.find('AssignFieldModal');
-        const modalProps = {
-            data: modal.prop('data'),
-            attributes: modal.prop('attributes'),
-            columns: modal.prop('columns'),
-            fieldsMap: modal.prop('fieldsMap'),
-        };
-        expect(modalProps).toEqual(baseParams);
+        expect(component.find('AssignFieldModal').props()).toEqual(
+            expect.objectContaining({
+                data: [{ Name: 'John', Age: 25 }, { Name: 'Marie', Age: 30 }],
+                attributes: {
+                    name: { required: true },
+                    age: {},
+                },
+                columns: ['Name', 'Age'],
+                fieldsMap: {
+                    name: 'Name',
+                    age: 'Age',
+                },
+            }),
+        );
     });
-    it('should prepare assign fields data using the passed params', () => {
-        const params = { ...baseParams, matchField: '' };
-        mount(<StepThree {...params} />);
+    it('should call getAssignFieldsData with the right data', () => {
+        mount(<StepThree {...baseParams} matchField="" />);
         expect(getAssignFieldsData).toHaveBeenCalledWith({
-            fieldsMap: baseParams.fieldsMap,
-            attributes: baseParams.attributes,
+            fieldsMap: {
+                name: 'Name',
+                age: 'Age',
+            },
+            attributes: {
+                name: { required: true },
+                age: {},
+            },
             matchField: '',
         });
     });
+    it('should pass only the first three elements of data prop to AssignFieldModal', () => {
+        const onAssignFieldFn = jest.fn();
+        const component = mount(<StepThree onAssignField={onAssignFieldFn} {...baseParams} />);
+        expect(component.find('AssignFieldModal').props()).toEqual(
+            expect.objectContaining({
+                attributes: {
+                    name: { required: true },
+                    age: {},
+                },
+                columns: ['Name', 'Age'],
+                onAssignField: onAssignFieldFn,
```

Comment:
Here you are not checking data prop. This test is meant to test ```const previewData = data.slice(0, 3);``` this code, that the data passed is only with the first 3 items, so you need to pass a data with more than 3 elements and then check only the data prop passed to AssignFieldModal